test/utilities.ts

```diff
@@ -334,6 +334,27 @@ export interface CommitEventArgs {
     amount: BigNumberish
     commitType: BigNumberish
 }
+
+/**
+ * Creates a commit and returns the event arguments for it
+ * @param poolAddresses array of LeveragedPool addresses to upkeep
+ * @param poolKeeper the PoolKeeper object
+ * @param l2Encoder the L2Encoder object for encoding function parameters
+ * @param signer the ethers Signer object (optional)
+ */
+export const performUpkeep = async (
+    poolAddresses: string[],
+    poolKeeper: PoolKeeper,
+    l2Encoder: L2Encoder,
+    signer?: Signer
+): Promise<any> /*Promise<CommitEventArgs>*/ => {
+    const encodedArgs = await l2Encoder.encodePerformUpkeepParams(poolAddresses)
+    signer = signer ? signer : (await ethers.getSigners())[0]
+    return await poolKeeper
```

Comment:
As this returns `Promise<any>`, we might not need to drive the final future?

```suggestion
    return poolKeeper
```

I've tested this locally and we seem to have to drive the inner futures (which makes sense as they're needed as parameters directly at the call site). The suggested patch I've left seems to work fine locally though (although I'd like other opinions also).